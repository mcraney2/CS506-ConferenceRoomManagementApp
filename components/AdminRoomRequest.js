import axios from 'axios';
import React, { Component } from 'react';
import { SafeAreaView, Text, View } from 'react-native';
import { FlatList } from 'react-native-gesture-handler';
import Button from './Button'

class AdminRoomRequest extends Component {
    constructor(props) {
        super(props);
        this.state = {  }
    }
    acceptRequest(id, refresh) {
        console.log("Send database 'accept' message to add to calendar and then database request to remove this from requests");
        console.log(id);
            //console.log(room,startTime, endTime, currentTime);
        const request = JSON.stringify(
              { 
                adminid:1,
                requestid: id
    
            });
        axios.post('http://10.0.2.2:8000/room_mgmt/admin/requests/process/', request)
            .then(function (response) {
              //console.log(response);
            })
            .catch(function (error) {
              console.log(error);
            })
            .then(function() {
              refresh();
            
  
            });
        }
    
    denyRequest(id, refresh) {
        console.log("Send database 'deny' request and simply remove request");
        console.log(id);
        const request = JSON.stringify(
            { 
              adminid:1,
              requestid: id
  
          });
        axios.put('http://10.0.2.2:8000/room_mgmt/admin/requests/process/',request)
        .then(function (response) {
            //console.log(response);
          })
          .catch(function (error) {
            console.log(error);
          })
          .then(function() {
            refresh();
          

          });
        }
    
    getDate(date) {
          
          
          var year = date.getFullYear();
          var month = date.getMonth() + 1;
          var day = date.getDate();
         
          var total = month + '/' + day + '/' + year;
          return total;
      }
    getTime(date) {
        
        
        var hour = date.getHours();
        var minutes = date.getMinutes();
        if (parseInt(minutes, 10) < 10) {
            minutes = '0' + minutes;
        }
        var total = hour + ':' + minutes;

        return total;
    }
     renderItem = ({ item }) => (
      // start = new Date(this.props.startTime);
      // end = new Date(this.props.endTime);
      // date = this.getDate(start);
      // startTime = this.getTime(start);
      // endTime = this.getTime(end);
      <View>
              <Text>Group: {item.group}</Text> 
              <Text>Event: {item.name}</Text>
              <Text>Date: {this.getDate(new Date(item.starttime))}</Text>
              <Text testID = 'startTime'>Start Time: {this.getTime(new Date(item.starttime))}</Text>
              <Text testID = 'endTime'>End Time: {this.getTime(new Date(item.endtime))}</Text>
              <Text>Reason: {item.reason}</Text>
              {/* <Text testID = 'conflicts'>Conflicts: {this.props.conflicts}</Text> */}
              <Button handleClick = {() =>this.acceptRequest(item.requestid,this.props.refresh)}
                      label = 'Accept'
                      />
              <Button handleClick = {() => this.denyRequest(item.requestid, this.props.refresh)}
                      label = 'Deny'
                      />
            </View>
    );
    render() { 
        //console.log(this.props);
        //console.log('start time', this.props.startTime);
        let start = new Date(this.props.startTime);
        let end = new Date(this.props.endTime);
        date = this.getDate(start);
        startTime = this.getTime(start);
        endTime = this.getTime(end);
        return (  
            // <View>
            //   <Text>Group: {this.props.group}</Text> 
            //   <Text>Event: {this.props.event}</Text>
            //   <Text>Date: {date}</Text>
            //   <Text testID = 'startTime'>Start Time: {startTime}</Text>
            //   <Text testID = 'endTime'>End Time: {endTime}</Text>
            //   <Text>Reason: {this.props.reason}</Text>
            //   {/* <Text testID = 'conflicts'>Conflicts: {this.props.conflicts}</Text> */}
            //   <Button handleClick = {() =>this.acceptRequest(this.props.id,this.props.refresh)}
            //           label = 'Accept'
            //           />
            //   <Button handleClick = {() => this.denyRequest(this.props.id, this.props.refresh)}
            //           label = 'Deny'
            //           />
            // </View>

            <SafeAreaView>
              <FlatList
                  data = {this.props.roomList}
                  renderItem = {this.renderItem}
                  keyExtractor = {item => item.requestid.toString()}
              />
            </SafeAreaView>
        );
    }
}
 
export default AdminRoomRequest;