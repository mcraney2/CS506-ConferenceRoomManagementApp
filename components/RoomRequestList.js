
import React, { Component } from 'react';
import {StyleSheet, Text, TouchableOpacityBase, View} from 'react-native'
import AdminRoomRequest from './AdminRoomRequest'
import jsonData from '../dummy-data'
import Accordion from 'react-native-collapsible/Accordion'
import axios from 'axios'


class RoomRequestList extends Component {
    constructor(props) {
        super(props);
        this.state = {  
            requests: [],
            activeSections : []
        }
    }
    componentDidMount() {
      //console.log("Send database request to get requests")
      this.getRequests();
        
    }
    getRequests(){
      //console.log('Get requests');
      const request = JSON.stringify(
        { 
          adminid:1,
          groupid: 1

      });
        axios.get('http://10.0.2.2:8000/room_mgmt/admin/requests/view/')
        .then(response => {
            this.setState({requests: response.data.requestlist})
        })
        .catch(function(error) {
            console.log(error)
        })
    }

       _renderHeader = section => {
        return (
          <View style={styles.header}>
            <Text style={styles.headerText}>Room: {section.room}</Text>
          </View>
        );
      };
      _renderContent = section => {
        return (
          <View style={styles.content}>
            {/* <AdminRoomRequest room = {section.room} startTime = {section.starttime} endTime = {section.endtime} group = {section.group} event = {section.name} reason = {section.reason} id = {section.requestid} refresh = {this.getRequests.bind(this)}/> */}
            <AdminRoomRequest roomList = {section.requests} refresh = {this.getRequests.bind(this)}/>
          </View>
        );
      };
      _updateSections = activeSections => {
        this.setState({ activeSections });
      };



      //  format Array[
      //   Object {
      //     room: X
      //     requests: []
      //   }
      // ]
      createArray(requests) {
        let arr = [];
        for (let i = 0; i < requests.length; i++) {
          let room = requests[i].room;
          let pushed = false;
          for (let j = 0; j < arr.length; j++) {
            if (arr[j].room === room) {
              arr[j].requests.push(requests[i]);
              pushed = true;
            }
          }
          if (!pushed) {
            arr.push({room : room, requests:[requests[i]]})
          }
        }
        
        return arr;
      }
    render() { 
      //console.log(DATA);
      //console.log(this.state.requests);
      let array = this.createArray(this.state.requests);
        return (  
            
                    // <Accordion
                    //     sections={this.state.requests}
                    //     activeSections={this.state.activeSections}
                    //     //renderSectionTitle={this._renderSectionTitle}
                    //     renderHeader={this._renderHeader}
                    //     renderContent={this._renderContent}
                    //     onChange={this._updateSections}
                    //  />
                    <Accordion
                    sections={array}
                    activeSections={this.state.activeSections}
                    //renderSectionTitle={this._renderSectionTitle}
                    renderHeader={this._renderHeader}
                    renderContent={this._renderContent}
                    onChange={this._updateSections}
                 />
                


            
        );
    }
}
const styles = StyleSheet.create({
    container: {
      flex: 1,
      flexDirection: 'row',
      alignItems: 'center',
      //justifyContent: 'center',
    },
    headerText: {
        fontSize: 22,
        fontWeight: 'bold'
    }
})
export default RoomRequestList;