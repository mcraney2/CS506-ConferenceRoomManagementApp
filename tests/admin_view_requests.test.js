import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AdminRoomRequest from '../components/AdminRoomRequest'
import RoomRequestList from '../components/RoomRequestList'
test('Renders Admin Room Request snapshot as expected', () => {
    const tree = renderer.create(<AdminRoomRequest/>).toJSON();
    expect(tree).toMatchSnapshot();
});
test('Renders Room Request List snapshot as expected', () => {
    const tree = renderer.create(<RoomRequestList/>).toJSON();
    expect(tree).toMatchSnapshot();
});

