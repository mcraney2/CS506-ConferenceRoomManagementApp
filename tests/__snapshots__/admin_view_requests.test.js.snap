// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Admin Room Request snapshot as expected 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTScrollView
    collapsable={false}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`Renders Room Request List snapshot as expected 1`] = `<View />`;
